server/models/testing/test-assignment.js

```diff
@@ -2,17 +2,51 @@ const mongoose = require('mongoose');
 
 const { Schema } = mongoose;
 
+const ASSIGNMENT_STATUS_PENDING = 'PENDING';
+const ASSIGNMENT_STATUS_PASSED = 'PASSED';
+const ASSIGNMENT_STATUS_EXPIRED = 'EXPIRED';
+
+const TYPE_TRAINING_TEST = 'TRAINING';
+const TYPE_PRIMARY_TEST = 'PRIMARY';
+
 const testAssignmentSchema = new Schema({
   groupId: { type: Schema.Types.ObjectId, ref: 'Group', index: true },
   studentId: { type: Schema.Types.ObjectId, ref: 'User', index: true },
   name: { type: String, required: true },
   tags: { type: [{ type: String }], index: true, required: true },
   assignDate: { type: Date, required: true },
+  timeToPass: { type: Number, required: true, default: 1200000 },
+  deadline: { type: Date },

```

Comment:
We don't use type: Date, consider using type: Number to store deadline in miliseconds.

Reply:
Look at other deadline fields for better understanding